src/lib/io/sol-iio.c

```diff
@@ -800,6 +866,12 @@ sol_iio_open(int device_id, const struct sol_iio_config *config)
         device->buffer_enabled, device->manual_triggering, device->trigger_name,
         device->trigger_id, device->mount_matrix);
 
+    entry = calloc(1, sizeof(struct iio_opened_device));
+    SOL_NULL_CHECK_GOTO(entry, error);
+    entry->device = device;
+    entry->config = config;
```

Comment:
`sol_iio_open` shouldn't take ownership of config struct. I guess that use on `iio/nodes.c` is ok (as it saves config struct), but this 'config to open' structs are supposed to be short lived. If we need to keep them here, we need to copy it.